src/testdrive/src/action/kafka/verify.rs

```diff
@@ -109,10 +109,23 @@ async fn get_topic(
     topic_field: &str,
     state: &mut State,
 ) -> Result<String, anyhow::Error> {
-    let query = format!("SELECT {} FROM mz_catalog_names JOIN mz_kafka_sinks ON global_id = sink_id WHERE name = $1", topic_field);
+    let query = format!(
+        "SELECT {} FROM mz_sinks JOIN mz_kafka_sinks \
+        ON mz_sinks.id = mz_kafka_sinks.sink_id \
+        JOIN mz_schemas s ON s.id = mz_sinks.schema_id \
+        LEFT JOIN mz_databases d ON d.id = s.database_id \
+        WHERE d.name = $1 \
+        AND s.name = $2 \
+        AND mz_sinks.name = $3",
+        topic_field
+    );
+    let sink_fields: Vec<&str> = sink.split('.').collect();
```

Comment:
Whew, it's so much better to confine this kind of logic to testdrive. This won't allow us to name sinks that contain `.` in their names, but since it's just in testdrive, that's 100% fine.

Reply:
Yeah the better solution would be to refactor all the testdrive tests to not use the dot notation and either take in multiple separate parameters, but I wasn't sure whether it's worth it.

Alternatively, we could always use the `materialize.public` schema (or not even specify a schema in the query) and not take that an input anymore in `kafka-verify`, since it doesn't look like any test uses a different schema to date.

Happy to do either of the above - let me know!

Reply:
Hah, those two solutions were exactly what I had in mind too! My sense is it’s probably not worth the time. Plenty of bigger, user-facing fish to fry.